src/gui/properties/peerlistwidget.cpp

```diff
@@ -262,44 +262,54 @@ void PeerListWidget::updatePeerCountryResolutionState()
+    BitTorrent::Torrent *torrent = m_properties->getCurrentTorrent();
     if (!torrent) return;
 
-    QMenu *menu = new QMenu(this);
+    auto *menu = new QMenu(this);
     menu->setAttribute(Qt::WA_DeleteOnClose);
+    menu->setToolTipsVisible(true);
 
-    // Add Peer Action
-    // Do not allow user to add peers in a private torrent
-    if (!torrent->isQueued() && !torrent->isChecking() && !torrent->isPrivate())
+    QAction *addNewPeer = menu->addAction(UIThemeManager::instance()->getIcon("user-group-new"), tr("Add a new peer...")
+        , this, [this, torrent]()
     {
-        menu->addAction(UIThemeManager::instance()->getIcon("user-group-new"), tr("Add a new peer...")
-            , this, [this, torrent]()
+        const QVector<BitTorrent::PeerAddress> peersList = PeersAdditionDialog::askForPeers(this);
+        const int peerCount = std::count_if(peersList.cbegin(), peersList.cend(), [torrent](const BitTorrent::PeerAddress &peer)
         {
-            const QVector<BitTorrent::PeerAddress> peersList = PeersAdditionDialog::askForPeers(this);
-            const int peerCount = std::count_if(peersList.cbegin(), peersList.cend(), [torrent](const BitTorrent::PeerAddress &peer)
-            {
-                return torrent->connectPeer(peer);
-            });
-            if (peerCount < peersList.length())
-                QMessageBox::information(this, tr("Adding peers"), tr("Some peers cannot be added. Check the Log for details."));
-            else if (peerCount > 0)
-                QMessageBox::information(this, tr("Adding peers"), tr("Peers are added to this torrent."));
+            return torrent->connectPeer(peer);
         });
-    }
+        if (peerCount < peersList.length())
+            QMessageBox::information(this, tr("Adding peers"), tr("Some peers cannot be added. Check the Log for details."));
+        else if (peerCount > 0)
+            QMessageBox::information(this, tr("Adding peers"), tr("Peers are added to this torrent."));
+    });
+    QAction *copyPeers = menu->addAction(UIThemeManager::instance()->getIcon("edit-copy"), tr("Copy IP:port")
+        , this, &PeerListWidget::copySelectedPeers);
+    menu->addSeparator();
+    QAction *banPeers = menu->addAction(UIThemeManager::instance()->getIcon("user-group-delete"), tr("Ban peer permanently")
+        , this, &PeerListWidget::banSelectedPeers);
+
+    // disable actions
+    const auto disableAction = [](QAction *action, const QString &tooltip)
+    {
+        action->setEnabled(false);
+        action->setToolTip(tooltip);
+    };
+
+    if (torrent->isPrivate())
+        disableAction(addNewPeer, tr("Cannot add peer to a private torrent"));
+    else if (torrent->isChecking())
+        disableAction(addNewPeer, tr("Cannot add peer when the torrent is checking"));
+    else if (torrent->isQueued())
+        disableAction(addNewPeer, tr("Cannot add peer when the torrent is queued"));
```

Comment:
We can add multiple peers at once so i'd change "peer" -> "peers" in all these strings.

Reply:
In my opinion, "can't add peer" implies that it "can't add _multiple_ peers" either.

Reply:
peer(s)?

Reply:
The menu item is "Add a new peer..." so I suppose the tooltip should follow (using singular nouns).

Reply:
Or change the menu item to "Add new peer(s)...", then it will match.

Reply:
>Or change the menu item to "Add new peer(s)...", then it will match.

Changed to plural.